File: src/code/java/binary_search/greedy_minimum.java
public int fn(int[] arr) {
    int left = MINIMUM_POSSIBLE_ANSWER;
    int right = MAXIMUM_POSSIBLE_ANSWER;

    while (left <= right) {
        int mid = (left + right) / 2;

        if (check(mid)) {
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }

    return left;
}

File: src/code/ruby/binary_search/greedy_minimum.rb
def fn(arr)
  left  = MINIMUM_POSSIBLE_ANSWER
  right = MAXIMUM_POSSIBLE_ANSWER

  while left <= right
    mid = (left + right) / 2 # int div

    if check(mid)
      right = mid - 1
    else
      left = mid + 1
    end
  end

  left
end

def check(x)
  BOOLEAN
end